champions/src/main/java/me/mykindos/betterpvp/champions/champions/skills/skills/assassin/passives/SiphoningStrikes.java

```diff
@@ -0,0 +1,112 @@
+package me.mykindos.betterpvp.champions.champions.skills.skills.assassin.passives;
+
+
+import java.util.HashSet;
+import java.util.WeakHashMap;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import me.mykindos.betterpvp.champions.Champions;
+import me.mykindos.betterpvp.champions.champions.ChampionsManager;
+import me.mykindos.betterpvp.champions.champions.skills.Skill;
+import me.mykindos.betterpvp.champions.champions.skills.types.PassiveSkill;
+import me.mykindos.betterpvp.core.combat.events.CustomDamageEvent;
+import me.mykindos.betterpvp.core.components.champions.Role;
+import me.mykindos.betterpvp.core.components.champions.SkillType;
+import me.mykindos.betterpvp.core.framework.updater.UpdateEvent;
+import me.mykindos.betterpvp.core.listener.BPvPListener;
+import me.mykindos.betterpvp.core.utilities.UtilTime;
+import org.bukkit.entity.Player;
+import org.bukkit.event.EventHandler;
+import org.bukkit.event.EventPriority;
+import org.bukkit.event.Listener;
+import org.bukkit.event.entity.EntityDamageEvent.DamageCause;
+
+@Singleton
+@BPvPListener
+public class SiphoningStrikes extends Skill implements PassiveSkill, Listener {
+
+    private final WeakHashMap<Player, Double> repeat = new WeakHashMap<>();
+    private final WeakHashMap<Player, Long> last = new WeakHashMap<>();
+
+    @Inject
+    public SiphoningStrikes(Champions champions, ChampionsManager championsManager) {
+        super(champions, championsManager);
+    }
+
+
+    @Override
+    public String getName() {
+        return "Siphoning Strikes";
+    }
+
+    private double healthIncrement;
+    private double duration;
+
+    @Override
+    public String[] getDescription(int level) {
+
+        return new String[]{
+                "Each time you attack, you siphon health",
+                "from the enemy, starting at <stat>" + ((healthIncrement * 100) * 2) + "%</stat>",
+                "and increasing by <val>" + (healthIncrement * 100) + "%</val> for each hit",
+                "",
+                "You can gain up to <val>" + (20+(level * (healthIncrement * 100))) + "%</val> health per hit",
+                "",
+                "Not attacking for <stat>" + duration + "</stat> seconds resets",
+                "your bonus health back to <stat>" +((healthIncrement * 100) * 2)+"%"};
+    }
+
+    @Override
+    public Role getClassType() {
+        return Role.ASSASSIN;
+    }
+
+    @Override
+    public SkillType getType() {
+        return SkillType.PASSIVE_A;
+    }
+
+
+    @EventHandler(priority = EventPriority.HIGH)
+    public void onDamage(CustomDamageEvent event) {
+        if (event.getCause() != DamageCause.ENTITY_ATTACK) return;
+        if (!(event.getDamager() instanceof Player damager)) return;
+        if (!championsManager.getRoles().hasRole(damager, Role.ASSASSIN)) return;
```

Comment:
This isn't really needed if you're already checking if their level is greater than 0 below this. Players who aren't wearing the role will return 0 anyway.